src/jvm/main/org/jetbrains/kotlinx/lincheck/Reporter.kt

```diff
@@ -136,18 +161,29 @@ internal fun StringBuilder.appendExecutionScenario(scenario: ExecutionScenario):
 }
 
 internal fun StringBuilder.appendFailure(failure: LincheckFailure): StringBuilder {
+    val results: ExecutionResult? = (failure as? IncorrectResultsFailure)?.results
```

Comment:
In Reporter and TraceReporter, please document everything you touch. These files are hard to read, so we need improve them.